lms/resources/lti_launch.py

```diff
@@ -38,31 +38,10 @@ def username():
             username_hash_object.update(self.h_provider_unique_id.encode())
             return username_hash_object.hexdigest()[:30]
 
-        def display_name():
-            """Return the h display name for the current request."""
-            params = self._request.parsed_params
-
-            display_name = params.get("lis_person_name_full", "").strip()
-
-            if not display_name:
-                given_name = params.get("lis_person_name_given", "").strip()
-                family_name = params.get("lis_person_name_family", "").strip()
-
-                display_name = " ".join((given_name, family_name)).strip()
-
-            if not display_name:
-                return "Anonymous"
-
-            # The maximum length of an h display name.
-            display_name_max_length = 30
-
-            if len(display_name) <= display_name_max_length:
-                return display_name
-
-            return display_name[: display_name_max_length - 1].rstrip() + "…"
-
         return HUser(
-            authority=self._authority, username=username(), display_name=display_name()
+            authority=self._authority,
+            username=username(),
+            display_name=self._request.lti_user.display_name,
```

Comment:
We no longer need to generate a `display_name` when we create an `HUser`. It's now generated earlier, when we create the `LTIUser` (in `LaunchParamsAuthSchema`)